README.md

```diff
@@ -2,33 +2,651 @@
+    query?: SelectedQueryMaybeVariablesFunc,
+    configQuery?: string,
+    processor?: DataProcessor,
+    view?: React.FunctionComponent<any>
+}
+```
+
+---
+
+#### `addMacro(name: string, obj: ComponentDefinition): void`
+
+#### `addPart(name: string, obj: ComponentDefinition): void`
+
+#### `addLayout(name: string, obj: ComponentDefinition): void`
+
+#### `addCPage(name: string, obj: ComponentDefinition): void`
+
+#### `addContentType(name: string, obj: ComponentDefinition): void`
+
+#### `addComponent(name: string, obj: ComponentDefinition): void`
+
+Saves the component definition in ComponentRegistry by name.
+
+> **NOTE:** <a id="add-comp"></a>`addComponent` is used for defining general types of Enonic XP components by `nextjs-adapter` so you don't
+> need to do it manually. Overriding default setup may break Enonic XP integration!
+
+| Argument | Description                       |
+|----------|-----------------------------------|
+| `name`   | Component name                    |
+| `obj`    | [Component definition](#comp-def) |
+
+Usage:
+
+```tsx
+import {ComponentRegistry} from '@enonic/nextjs-adapter';
+
+const definition = {}
+
+ComponentRegistry.addMacro('macro-name', definition);
+```
+
+---
+
+#### `getMacro(name: string): ComponentDefinition | undefined`
+
+#### `getPart(name: string): ComponentDefinition | undefined`
+
+#### `getLayout(name: string): ComponentDefinition | undefined`
+
+#### `getPage(name: string): ComponentDefinition | undefined`
+
+#### `getContentType(name: string): ComponentDefinition | undefined`
+
+#### `getComponent(name: string): ComponentDefinition | undefined`
+
+Gets the component definition stored in ComponentRegistry by its name.
+
+> **NOTE:** Read [addComponent](#add-comp) note before using `getComponent`.
+
+| Argument | Description    |
+|----------|----------------|
+| `name`   | Component name |
+
+Usage:
+
+```tsx
+import {ComponentRegistry} from '@enonic/nextjs-adapter';
+
+const definition = ComponentRegistry.getMacro('macro-name');
+```
+
+Response type: [component definition](#comp-def)
+
+---
+
+#### `getMacros(): ComponentDefinition[]`
+
+#### `getParts(): ComponentDefinition[]`
+
+#### `getLayouts(): ComponentDefinition[]`
+
+#### `getPages(): ComponentDefinition[]`
+
+#### `getContentTypes(): ComponentDefinition[]`
+
+#### `getComponents(): ComponentDefinition[]`
+
+Gets all component definition stored in ComponentRegistry.
+
+
+> **NOTE:** Read [addComponent](#add-comp) note before using `getComponents`.
+
+Usage:
+
+```tsx
+import {ComponentRegistry} from '@enonic/nextjs-adapter';
+
+const macros = ComponentRegistry.getMacros();
+```
+
+Response type: List of [component definitions](#comp-def)
+
+### `UrlProcessor`
+
+Helper singleton for processing URLs.
+
+---
+
+#### <a id="process"></a>`static process(url: string, meta: MetaData): string`
+
+Processes the absolute URL to become relative for the current viewer, while keeping in mind Next.js assets and Enonic XP binary content
```

Comment:
Unnecessary linebreak?

Reply:
Doesn't affect the output, but easier to navigate the source